hidden=until-found post (beta 102) (#2708)
GoogleChrome/developer.chrome.com: site/en/blog/hidden-until-found/index.md

---
title: 'Making collapsed content accessible with hidden=until-found'
description: >
  How this new attribute value can ensure that content within accordion sections can be found and linked to. 
layout:
  'layouts/blog-post.njk'

> date: 2022-04-27

Author: ```suggestion
date: 2022-04-28
```

authors:
  - jarhar
hero: 'image/kheDArv5csY6rvQUJDbWRscckLr1/bt6eW4gluWeTMRNfvKa6.jpg'
alt: >
  A cat hiding behind a curtain.
tags:
  - chrome
  - html
---

> Collapsing sections of content is a common UI pattern, sometimes described as [an accordion](https://www.smashingmagazine.com/2017/06/designing-perfect-accordion-checklist/), however content hidden in the collapsed sections becomes impossible to search using a find-in-page search. Also, it isn't possible to link to [text fragments](https://web.dev/text-fragments/) inside collapsed regions.

Reviewer: Instead of phrasing it like "Collapsing sections of content is a common UI pattern, sometimes described as an accordion, however content..."
How about like this? "Collapsing sections of content, sometimes described as an accordion, is a common UI pattern. However, content..."

> Collapsing sections of content, sometimes described as [an accordion](https://www.smashingmagazine.com/2017/06/designing-perfect-accordion-checklist/), is a common UI pattern. However, content hidden in the collapsed sections becomes impossible to search using a find-in-page search. Also, it isn't possible to link to [text fragments](https://web.dev/text-fragments/) inside collapsed regions.

Reviewer: is a common UI pattern -> are a common UI pattern

Reviewer: ...I think? If this grammar is already correct then "is" is fine, it just kind of seems plural because of the "s" in "sections"

Author: ooh, I feel like it could be correct either way, for me the "is"relates to collapsing rather than sections. On discussing with TW I'm rewording slightly which I hope works better.

> Collapsing sections of content, sometimes described as [an accordion](https://www.smashingmagazine.com/2017/06/designing-perfect-accordion-checklist/), is a common UI pattern. However, content hidden in the collapsed sections becomes impossible to search using a find-in-page search. Also, it isn't possible to link to [text fragments](https://web.dev/text-fragments/) inside collapsed regions.

Author: ```suggestion
Collapsing content sections, sometimes described as [an accordion](https://www.smashingmagazine.com/2017/06/designing-perfect-accordion-checklist/), are a common UI pattern. However, content hidden in the collapsed sections becomes impossible to search using a find-in-page search. Also, it isn't possible to link to [text fragments](https://web.dev/text-fragments/) inside collapsed regions.
```

> The `hidden=until-found` HTML attribute and `beforematch` event can solve these problems. By adding `hidden=until-found` to the container for your hidden content, you make it possible for the browser to match text in that hidden region, and reveal the section if a match is found.

Reviewer: Replace "match text in that hidden region" with "search text in that hidden region"?
